Polymorphic type reference error during build

### System Information

This error occurs after make j*:

```

Scanning dependencies of target opencv_perf_imgproc
[ 37%] Building CXX object modules/imgproc/CMakeFiles/opencv_perf_imgproc.dir/perf/opencl/perf_3vs4.cpp.o
[ 37%] Building CXX object modules/cudaarithm/CMakeFiles/opencv_test_cudaarithm.dir/test/test_event.cpp.o
[ 37%] Building CXX object modules/cudaarithm/CMakeFiles/opencv_perf_cudaarithm.dir/perf/perf_element_operations.cpp.o
/home/ubuntu/opencv_contrib/modules/cudaarithm/test/test_event.cpp: In member function ‘void opencv_test::{anonymous}::AsyncEvent_Timing_Test::UnsafeTestBody()’:
/home/ubuntu/opencv_contrib/modules/cudaarithm/test/test_event.cpp:94:30: warning: catching polymorphic type ‘class cv::Exception’ by value [-Wcatch-value=]
   94 |         catch (cv::Exception ex) {


```

The warning message you're seeing is related to catching a polymorphic type by value instead of by reference. In C++, it's a good practice to catch exceptions by reference rather than by value to preserve the exception type information and to avoid object slicing.

Here's how you could fix this warning:

Change this line:
`catch (cv::Exception ex) {`

To:
`catch (const cv::Exception& ex) {`



### Detailed description

By making this change, you're catching the exception by reference, which should eliminate the warning message. Now go back to your source code, find the file test_event.cpp, locate line 94, and make the suggested change. Save the file and re-compile your project. This should resolve the warning you were seeing.


### Steps to reproduce

make j*

### Issue submission checklist

- [X] I report the issue, it's not a question
- [X] I checked the problem with documentation, FAQ, open issues, forum.opencv.org, Stack Overflow, etc and have not found any solution
- [X] I updated to the latest OpenCV version and the issue is still there
- [ ] There is reproducer code and related data files (videos, images, onnx, etc)

--- a/modules/cudaarithm/test/test_event.cpp
+++ b/modules/cudaarithm/test/test_event.cpp
@@ -91,7 +91,7 @@ CUDA_TEST_P(AsyncEvent, Timing)
             const double elTimeMs = Event::elapsedTime(startEvent, stopEvent);
             ASSERT_GT(elTimeMs, 0);
         }
-        catch (cv::Exception ex) {
+        catch (const cv::Exception& ex) {
             failed = true;
         }
         ASSERT_EQ(failed, shouldFail.at(i));

--- a/modules/datasets/src/tinyxml2/tinyxml2.h
+++ b/modules/datasets/src/tinyxml2/tinyxml2.h
@@ -212,7 +212,7 @@ template <class T, int INIT>
 class DynArray
 {
 public:
-    DynArray< T, INIT >() {
+    DynArray() {
         _mem = _pool;
         _allocated = INIT;
         _size = 0;

--- a/modules/xphoto/src/oilpainting.cpp
+++ b/modules/xphoto/src/oilpainting.cpp
@@ -58,7 +58,7 @@ private:
     int dynRatio;
 
 public:
-    ParallelOilPainting<Type>(Mat& img, Mat &d, Mat &iLuminance, int r,int k) :
+    ParallelOilPainting(Mat& img, Mat &d, Mat &iLuminance, int r,int k) :
         imgSrc(img),
         dst(d),
         imgLuminance(iLuminance),

